x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/components/tables/meta_engines_table.test.tsx

```diff
@@ -0,0 +1,224 @@
+  const onChange = jest.fn();
+  const onDeleteEngine = jest.fn();
+
+  const data = [
+    {
+      name: 'test-engine',
+      created_at: 'Fri, 1 Jan 1970 12:00:00 +0000',
+      language: 'English',
+      isMeta: true,
+      document_count: 99999,
+      field_count: 10,
+      includedEngines: [
+        {
+          name: 'source-engine-1',
+        },
+        { name: 'source-engine-2' },
+      ],
+    } as EngineDetails,
+  ];
+  const pagination = {
+    pageIndex: 0,
+    pageSize: 10,
+    totalItemCount: 50,
+    hidePerPageOptions: true,
+  };
+  const props = {
+    items: data,
+    loading: false,
+    pagination,
+    onChange,
+    onDeleteEngine,
+  };
+
+  const DEFAULT_VALUES = {
+    myRole: {
+      canManageMetaEngines: false,
+    },
+    expandedSourceEngines: {},
+    hideRow: jest.fn(),
+    fetchOrDisplayRow: jest.fn(),
+  };
+
+  const resetMocks = () => {
+    jest.clearAllMocks();
+    setMockValues(DEFAULT_VALUES);
+  };
+
+  describe('basic table', () => {
+    let wrapper: ReactWrapper<any>;
+    let table: ReactWrapper<any>;
+
+    beforeAll(() => {
+      resetMocks();
+      wrapper = mountWithIntl(<MetaEnginesTable {...props} />);
+      table = wrapper.find(EuiBasicTable);
+    });
+
+    it('renders', () => {
+      expect(table).toHaveLength(1);
+      expect(table.prop('pagination').totalItemCount).toEqual(50);
+      expect(table.find(MetaEnginesTableNameColumnContent)).toHaveLength(1);
+
+      const tableContent = table.text();
+      expect(tableContent).toContain('Jan 1, 1970');
+      expect(tableContent).toContain('99,999');
+      expect(tableContent).toContain('10');
+
+      expect(table.find(EuiPagination).find(EuiButtonEmpty)).toHaveLength(5); // Should display 5 pages at 10 engines per page
+    });
+
+    it('contains engine links which send telemetry', () => {
+      const engineLinks = wrapper.find(EuiLinkTo);
+
+      engineLinks.forEach((link) => {
+        expect(link.prop('to')).toEqual('/engines/test-engine');
+        link.simulate('click');
+
+        expect(mockTelemetryActions.sendAppSearchTelemetry).toHaveBeenCalledWith({
+          action: 'clicked',
+          metric: 'engine_table_link',
+        });
+      });
+    });
+
+    it('triggers onPaginate', () => {
+      table.prop('onChange')({ page: { index: 4 } });
+      expect(onChange).toHaveBeenCalledWith({ page: { index: 4 } });
+    });
+  });
+
+  describe('loading', () => {
+    it('passes the loading prop', () => {
+      resetMocks();
+      const wrapper = mountWithIntl(<MetaEnginesTable {...props} loading />);
+
+      expect(wrapper.find(EuiBasicTable).prop('loading')).toEqual(true);
+    });
+  });
+
+  describe('noItemsMessage', () => {
+    it('passes the noItemsMessage prop', () => {
+      resetMocks();
+      const wrapper = mountWithIntl(<MetaEnginesTable {...props} noItemsMessage={'No items.'} />);
+      expect(wrapper.find(EuiBasicTable).prop('noItemsMessage')).toEqual('No items.');
+    });
+  });
```

Comment:
[Optional, **not** a change request] I noticed these as well as a few other tests were repeated between the EnginesTable and MetaEnginesTable - that is totally fine and not a big deal at all, but if you ever want to DRY out tests that are super similar between two components, consider a `test_helpers/` file which takes a wrapper and DRY's out asserts!

Example: https://github.com/elastic/kibana/blob/master/x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/components/analytics_tables/test_helpers/shared_columns_tests.tsx

I'd be happy to grab this in a later PR (along with the other DRYing out type comments I made in this review round) or push to this branch if you're interested!

Reply:
https://github.com/elastic/kibana/pull/96251/commits/6fcf91c44bc8ef14e41e5babfe49579e8714bc8a